packages/core/test/features/css-contains.spec.ts

```diff
@@ -0,0 +1,496 @@
+    it('should collect and namespace "container-name" and "container" decls', () => {
+        const { sheets } = testStylableCore(`
+            .a {
+                /* @decl(single) container-name: entry__con-a */
+                container-name: con-a;
+            
+                /* @decl(multi) container-name: entry__b entry__c */
+                container-name: b c;
+
+                /* @decl(shorthand) container: entry__d */
+                container: d;
+
+                /* @decl(shorthand+type) container: entry__e / normal */
+                container: e / normal;
+            }
+        `);
+
+        const { meta, exports } = sheets['/entry.st.css'];
+
+        shouldReportNoDiagnostics(meta);
+
+        // symbols
+        expect(CSSContains.get(meta, 'con-a'), 'con-a symbol').to.eql({
+            _kind: 'container',
+            name: 'con-a',
+            alias: 'con-a',
+            global: false,
+            import: undefined,
+        });
+        expect(CSSContains.get(meta, 'b'), 'b symbol').to.eql({
+            _kind: 'container',
+            name: 'b',
+            alias: 'b',
+            global: false,
+            import: undefined,
+        });
+        expect(CSSContains.get(meta, 'c'), 'c symbol').to.eql({
+            _kind: 'container',
+            name: 'c',
+            alias: 'c',
+            global: false,
+            import: undefined,
+        });
+        expect(CSSContains.get(meta, 'd'), 'd symbol').to.eql({
+            _kind: 'container',
+            name: 'd',
+            alias: 'd',
+            global: false,
+            import: undefined,
+        });
+        expect(CSSContains.get(meta, 'e'), 'e symbol').to.eql({
+            _kind: 'container',
+            name: 'e',
+            alias: 'e',
+            global: false,
+            import: undefined,
+        });
+
+        // JS exports
+        expect(exports.containers).to.eql({
+            'con-a': 'entry__con-a',
+            b: 'entry__b',
+            c: 'entry__c',
+            d: 'entry__d',
+            e: 'entry__e',
+        });
+    });
+    it('should preserve none container name', () => {
+        const { sheets } = testStylableCore(`
+            .a {
+                /* @decl(longhand) container-name: none */
+                container-name: none;
+
+                /* @decl(shorthand) container: none / normal */
+                container: none / normal;
+            }
+        `);
+
+        const { meta, exports } = sheets['/entry.st.css'];
+
+        shouldReportNoDiagnostics(meta);
+
+        // symbols
+        expect(CSSContains.get(meta, 'none'), 'no none symbol').to.eql(undefined);
+
+        // JS exports
+        expect(exports.containers).to.eql({});
+    });
+    it('should mark as global', () => {
+        const { sheets } = testStylableCore(`
+            .a {
+                /* @decl(use global) container-name: a b c d e */
+                container-name: a b c d e;
```

Comment:
so st-global override the local for each symbol?

Reply:
resolved in offline discussion: https://github.com/wix/stylable/pull/2744#issuecomment-1310035240